src/Package/Impl/Sql/DragDrop/DataObject.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Windows;
+using Microsoft.Common.Core;
+using Microsoft.Languages.Editor.DragDrop;
+
+namespace Microsoft.VisualStudio.R.Package.Sql.DragDrop {
+    public static class DataObject {
+        public static string GetPlainText(this IDataObject dataObject) {
+            var flags = dataObject.GetFlags();
+            if ((flags & DataObjectFlags.ProjectItems) != 0) {
+                var item = dataObject.GetProjectItems().FirstOrDefault();
+                if (item != null && Path.GetExtension(item.FileName).EqualsIgnoreCase(".r")) {
+                    return GetFileContent(item.FileName).Replace("'", "''");
```

Comment:
Oh, it wouldn't be legal R code. But it could still be in an .R file, since it's user input, so it can be anything at all. Highly unlikely, but we might want to check what happens so as to not end with weird SQL output that doesn't execute because it's not syntactically valid, just as a matter of general input sanitization.